timeseries/src/autogluon/timeseries/metrics/point.py

```diff
@@ -246,8 +245,11 @@ def compute_metric(
         if self._past_abs_seasonal_error is None:
             raise AssertionError("Call `save_past_metrics` before `compute_metric`")
 
-        mae_per_item = (y_true - y_pred).abs().groupby(level=ITEMID, sort=False).mean()
-        return self._safemean(mae_per_item / self._past_abs_seasonal_error)
+        num_items = len(self._past_abs_seasonal_error)
+        # Reshape abs errors into [num_items, prediction_length] to normalize per item without groupby
+        abs_errors = np.abs(y_true.values - y_pred.values).reshape([num_items, -1])
+        # We assume that items are in the same order in both arrays because predictor sorts by item_id
+        return self._safemean(abs_errors / self._past_abs_seasonal_error.values[:, None])
```

Comment:
This change is necessary to make sure that our MASE/SQL/RMSSE implementation aligns with GluonTS if missing values are present. For example, suppose that `prediction_length=100` and we have two items: 
- item A with 1 observed value in the forecast horizon
- item B with 100 observed values in the forecast horizon

In our old implementation with groupby, we'd first compute the mean for each item, and then take mean across items. This would assign 100x more weight to the single value for item A compared to the values in item B.

In the new implementation, each observed value in the forecast horizon is weighted equally (same as in GluonTS).

Reply:
One will need to get really creative to violate this assumption. Even if one uses a custom `TimeSeriesScorer` outside of the predictor, the following code will work because `predictor.predict` preserved the item_id order
```python
data = TimeSeriesDataFrame(...)  # load data with unsorted item_ids
predictions = TimeSeriesPredictor(...).fit(data).predict(data)  # `predictions` have the same item_id order as `data`
mase = MASE()(data, predictions, ...)  # this works
```
One will need to explicitly shuffle `predictions`, or shuffle `data` before calling the scorer to violate this assumption.

Reply:
OK so more generally, data and predictions should have the same index order. As long as it's documented clearly I'm OK.

Reply:
You are absolutely right, `TimeSeriesScorer` always assumes that predictions & data have the same index, otherwise an AssertionError is raised https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/metrics/abstract.py#L80. 

I guess I can completely remove the following comment
```python
# We assume that items are in the same order in both arrays because predictor sorts by item_id
```

Reply:
Sure. Also making it clearer in the docstrings would be great.

Reply:
Done :+1: